### PR TITLE
Add filtering

### PR DESCRIPTION
**Pull request by martynjames**
@dino-cikatic - Filtering addition to course discovery code...

### CODE REVIEW

**Comment by martynjames** [comment]
@dino-cikatic @dsego - Also see the change to the course discovery at https://github.com/edx/edx-platform/tree/mjames/course-discovery-test - this enables one to use the syntax 'org:edX' etc. to search course discovery results by 'org' and 'modes' - we can add more filters very easily as we see fit. I don't think we need those changes released, but at least it will demonstrate how the filtering is accessed from the calling javascript.

**Comment by martynjames** [comment]
@dino-cikatic, @marjev, @dsego  - Can you review this please?

File: search/views.py [hunk 1]
```diff
@@ -48,7 +60,7 @@ def do_search(request, course_id=None):
         http json response with the following fields
             "took" - how many seconds the operation took
             "total" - how many results were found
-            "max_score" - maximum score from these resutls
+            "max_score" - maximum score from these results
```
Comment (marjev) [inline search/views.py]: NIT: see line #156, same misspelling there

**Comment by marjev** [comment]
lgtm :+1:

Is there a reason why Travis didn't run tests for this PR?

**Comment by dsego** [comment]
:+1:

**Comment by dino-cikatic** [comment]
:+1:  looks good

**Comment by martynjames** [comment]
@marjev - Yes, travis is configured to only run tests for PRs against master

**Pull request merged**